src/community/security/keycloak/src/main/java/org/geoserver/security/keycloak/KeycloakUrlBuilder.java

```diff
@@ -43,7 +47,11 @@ KeycloakUrlBuilder client(String clientId) {
      */
     KeycloakUrlBuilder userByName(String userName) {
         users();
-        sb.append("?exact=true&username=").append(userName);
+        try {
+            sb.append("?exact=true&username=").append(URLEncoder.encode(userName, "UTF-8"));
```

Comment:
You can probably avoid the try / catch if you do something like:

```
sb.append("?exact=true&username=").append(URLEncoder.encode(userName, StandardCharsets.UTF_8));
```

Reply:
Good thinking, I changed it